azure-proto-core-test/SystemAssignedIdentityTests.cs

```diff
@@ -0,0 +1,299 @@
+            string value = "";
+            using (Stream stream = new MemoryStream())
+            {
+                using (StreamReader streamReader = new StreamReader(stream))
+                {
+                    var writer = new Utf8JsonWriter(stream);
+                    Serialize(writer, systemAssignedIdentity);
+                    stream.Seek(0, SeekOrigin.Begin);
+                    value = streamReader.ReadToEnd();
+                }
+            }
+            string actual = "{\"principalId\":\"de29bab1-49e1-4705-819b-4dfddceaaa98\",\"tenantId\":\"72f988bf-86f1-41af-91ab-2d7cd011db47\"}";
+            Assert.AreEqual(actual, value);
+        }
+
+        [TestCase]
+        public void TestSerializerEmptyIdentity()
+        {
+            SystemAssignedIdentity systemAssignedIdentity = new SystemAssignedIdentity();
+            string value = "";
+            using (Stream stream = new MemoryStream())
+            {
+                using (StreamReader streamReader = new StreamReader(stream))
+                {
+                    var writer = new Utf8JsonWriter(stream);
+                    Serialize(writer, systemAssignedIdentity);
+                    stream.Seek(0, SeekOrigin.Begin);
+                    value = streamReader.ReadToEnd();
+                }
+            }
+            string actual = "{\"principalId\":\"null\",\"tenantId\":\"null\"}";
+            Assert.AreEqual(actual, value);
+        }
+
+        [TestCase]
+        public void TestSerializerNullIdentity()
+        {
+            SystemAssignedIdentity systemAssignedIdentity = null;
+            using (Stream stream = new MemoryStream())
+            {
+                var writer = new Utf8JsonWriter(stream);
+                Assert.Throws<ArgumentNullException>(delegate { Serialize(writer, systemAssignedIdentity); });
+            }
+        }
+
+        [TestCase]
+        public void TestSerializerNullWriter()
+        {
+            SystemAssignedIdentity systemAssignedIdentity = new SystemAssignedIdentity();
+            using (Stream stream = new MemoryStream())
+            {
+                Assert.Throws<ArgumentNullException>(delegate { Serialize(null, systemAssignedIdentity); });
+            }
+        }
+
+        [TestCase]
+        public void TestEqualsBothIdentitiesNull()
+        {
+            SystemAssignedIdentity identity1 = null;
+            SystemAssignedIdentity identity2 = null;
+            Assert.IsTrue(Equals(identity1, identity2));
+        }
+
+        [TestCase]
+        public void TestEqualsOneIdentityNull()
+        {
+            SystemAssignedIdentity identity1 = null;
+            SystemAssignedIdentity identity2 = new SystemAssignedIdentity();
+            Assert.IsFalse(Equals(identity1, identity2));
+        }
+
+        [TestCase]
+        public void TestEqualsOtherIdentityNull()
+        {
+            SystemAssignedIdentity identity1 = new SystemAssignedIdentity();
```

Comment:
How do these tests differ from the equality tests above?  What is being tested in addition?